### PR TITLE
- [glib] use MesonToolchain

### PR DESCRIPTION
**Pull request by SSE4**
Specify library name and version:  **glib/all**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by jwillikers** [comment]
@SSE4 I'm seeing the same issue on macOS as you in my similar PR #11610. Does this require Conan Center to specify `tools.apple:sdk_path`?

**Review by prince-chrismc — APPROVED** [review]

File: recipes/glib/all/conanfile.py [hunk 1]
```diff
@@ -88,91 +92,108 @@ def requirements(self):
...
+            self.conf["tools.apple:sdk_path"] = apple_sdk_path(self)
+
+        tc = PkgConfigDeps(self)
+        tc.generate()
+        tc = AutotoolsDeps(self)
+        # bug? meson toolchain doesn't read CPPFLAGS
+        cppflags = tc.vars().get("CPPFLAGS")
+        tc.environment.append('CFLAGS', cppflags)
+        tc.environment.append('CXXFLAGS', cppflags)
+        # conan or meson bug? LIBPATH is ignored
```
Comment (prince-chrismc) [inline recipes/glib/all/conanfile.py:133]: Is it worth opening an issue in the main repo to track?

File: recipes/glib/all/conanfile.py [hunk 2]
```diff
@@ -88,91 +92,108 @@ def requirements(self):
...
+        ldflags = tc.vars().get("LDFLAGS")
+        ldflags = ldflags.replace("-LIBPATH", "/LIBPATH")
+        tc.environment.define('LDFLAGS', ldflags)
+        tc.generate()
+        # it's needed so MesonToolchain reads from AutotoolsDeps, should it be automatic?
```
Comment (prince-chrismc) [inline recipes/glib/all/conanfile.py:138]: Main repo feature/question issue?

Comment (jwillikers) [reply recipes/glib/all/conanfile.py:138]: Um, should you add the `VirtualBuildEnv` and `VirtualRunEnv` generators here instead?

**Review by jwillikers — CHANGES_REQUESTED** [review]

File: recipes/glib/all/conanfile.py (hunk 1 above)
Comment (franramirez688) [reply recipes/glib/all/conanfile.py:133]: I think that all those problems were resolved by https://github.com/conan-io/conan/pull/11557 (regarding the issue itself https://github.com/conan-io/conan/issues/11398).
Perhaps, it's worth starting to try out the new `MesonDeps` (Conan 1.51) generator for this case. See more information at https://docs.conan.io/en/latest/reference/conanfile/tools/meson/mesondeps.html.

File: recipes/glib/all/conanfile.py (hunk 2 above)
Comment (franramirez688) [reply recipes/glib/all/conanfile.py:138]: Same comment as https://github.com/conan-io/conan-center-index/pull/11053/files#r941237149

File: recipes/glib/all/conanfile.py (hunk 1 above)
Comment (SSE4) [reply recipes/glib/all/conanfile.py:133]: https://github.com/conan-io/conan/issues/11398

File: recipes/glib/all/conanfile.py (hunk 2 above)
Comment (SSE4) [reply recipes/glib/all/conanfile.py:138]: https://github.com/conan-io/conan/issues/10683

Comment (SSE4) [reply recipes/glib/all/conanfile.py:138]: @jwillikers why?

Comment (jwillikers) [reply recipes/glib/all/conanfile.py:138]: I wasn't sure if those would help, but probably not given the linked issues and such.

**Review by jwillikers — APPROVED** [review]

**Pull request closed without merging**